### DISCUSSION TITLE
Extending Database with more tables

### DISCUSSION THREAD

**Discussion by yannickgloster** [nextauthjs/next-auth#1745] [Help] [answered]
**How do I extend the database to have another table**

**I'm creating groups of users and am trying to create a database that contains groups and another one that has list of users connected to a groups**

**Feedback**

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by j-lee8** [comment]
Is it possible to intercept the database user insert and add additional fields? I'm using Prisma2 and have additional columns and I need to populate these on sign up.

**Reply by yannickgloster** [reply]
I managed to do this by using the JWT callback. I'm not sure that this is the correct way to do it though. 
```javascript
  callbacks: {
    async jwt(token, user, account, profile, isNewUser) {
      if (isNewUser) {
        try {
          const folder = await axios.post(
            "https://www.googleapis.com/drive/v3/files",
            {
              name: "SECURE",
              mimeType: "application/vnd.google-apps.folder",
            },
            {
              headers: {
                authorization: "Bearer " + account.accessToken,
              },
            }
          );
          await prisma.user.update({
            where: { id: user.id },
            data: { folderID: folder.data.id },
          });
        } catch (error) {
          console.log(error);
          console.log("Couldn't create folder");
        }
      }
      // Add access_token to the token right after signin
      if (account?.accessToken) {
        token.accessToken = account.accessToken;
      }

      return token;
    }
```

You could also maybe use the creatUser event:
```javascript
  events: {
    async createUser(message) {},
  }
```

It all just depends where you get that information from.

Probably the most correct way would be to fork the prisma database adapter and change the onSignup function.

**Reply by xcaeser** [reply]
the events is the right answer, thank you!

**Comment by yannickgloster** [comment, marked as answer]
I was able to do this by switching to using Prisma instead of TypeORM which allowed me to define my DB schema.
